routers/web/web.go

```diff
@@ -99,6 +99,18 @@ func Routes(sessioner func(http.Handler) http.Handler) *web.Route {
 		http.Redirect(w, req, path.Join(setting.StaticURLPrefix, "/assets/img/apple-touch-icon.png"), 301)
 	})
 
+	// the custom favicon should be seen when viewing files instead of the default one if it exists,
+	routes.Get("/favicon.ico", func(w http.ResponseWriter, req *http.Request) {
+		faviconPath := ""
+		res, err := http.Get(path.Join(setting.StaticURLPrefix, "/assets/img/favicon.ico"))
```

Comment:
It's better to only check this once and not make a request every time when `favicon.ico` is requested. We can use `os.Stat` to check if such file exists.

Reply:
There are a number of logo files that are known to exist, basically the output files here:

https://github.com/go-gitea/gitea/blob/fb08d2b3fd04dc8fc6cbd45fd341773b817c0857/build/generate-images.js#L74-L78

If browser accept redirecting ico to either svg or png, I think this is the solution we should go for as `favicon.ico` generally does not exist for our installations (but users could provide it for compatibilty)

Reply:
I think it may work to redirect to PNG, I'll give that a shot.

Reply:
Yeah, it's likely if PNG works, it will work with SVG as well, see https://caniuse.com/link-icon-svg for browser support.

Reply:
Another option might be to just generate a `ico` in above script, but that might be more complicated, and I think it should only be a last-resort option if those redirects don't work.